### DISCUSSION TITLE
If flux bootstrap can connect to git by proxy

### DISCUSSION THREAD

**Discussion by Xinyan272** [fluxcd/flux2#1422] [Proposals]
My problem:
In the company, the gitlab is private, and it is on a physical machine different from the cluster, and cannot be connected unless through a proxy. It means we have an intermediate server, which can connect to git, and the cluster can also connect to it, so the intermediate server will act as a proxy server for gitlab.

Since flux does not provide an option to use a proxy to connect to the git repository, we don't know how to use it.

Why can't flux bootstrap provide a variable PROXY like the git proxy of tekton's pipeline resource?

**Comment by richardh65** [comment]
Hi all,

We had the same issue. What we did was add --path to the bootstrap

flux bootstrap git --path=cluster/test 
and then in the 
cluster/test/flux-system
add a deploy patch file that had a proxy

extra-env-patch.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
   name: source-controller
   namespace: flux-system
spec:
  template:
  spec: 
    containers:
    - name: manager
      env:
          name: http_proxy
           value: <<ours>>
          name:https_proxy
           value: <<ours>>
          name: no_proxy
           value: "172.20.0.1, etc, etc"

then added a reference to that file in
kustomization.yaml
patchesStrategicMerge:
  - extra-env-patch.yaml

and bootstrap could reeach our gitrepo

hope that helps

**Comment by bread-cat** [comment]
Is it possible to run bootstrap with proxy settings instead of doing patch after provision?
